kale/pipeline/mida_trainer.py

```diff
@@ -0,0 +1,200 @@
+        self.best_estimator.fit(x_transformed[y.shape[0] :,], y)
+
+        return self
+
+    def _fit_mida(self, x_transformed, y, groups, transformer_params):
+        """Fit MIDA and estimator with the given data x_transformed and labels
+        Args:
+            x_transformed (array-like): Transformed data, shape (n_samples, n_features)
+            y (array-like): data labels, shape (n_labeled_samples,)
+            groups (array-like): group labels for the samples, shape (n_samples,)
+            transformer_params (dict): Parameters of the transformer
+
+        Returns:
+            self
+        """
+
+        mida_grid = ParameterGrid(self.mida_param_grid)
+        for mida_params in mida_grid:
+            mida = MIDA(**mida_params)
+            x_transformed = mida.fit_transform(x_transformed, y=y, groups=groups)
+            cv = 5
+            if np.unique(groups).shape[0] > 2:
+                cv = LeaveOneGroupOut().split(x_transformed[y.shape[0] :], y, groups[y.shape[0] :])
+            self._fit_estimator(x_transformed, y, mida_params, transformer_params, cv=cv)
+
+        return self
+
+    def _fit_estimator(self, x_transformed, y, mida_params, transformer_params, cv=5):
+        """Fit the estimator with the given data x_transformed and labels y
+        Args:
+            x_transformed (array-like): Transformed data, shape (n_samples, n_features)
+            y (array-like): data labels, shape (n_labeled_samples,)
+            mida_params (dict): Parameters of the MIDA
+            transformer_params (dict): Parameters of the transformer
+
+        Returns:
+            self
+        """
+
+        clf = GridSearchCV(self.estimator, self.estimator_param_grid, cv=cv)
+        clf.fit(x_transformed[y.shape[0] :,], y)
+        if self.best_score is None or clf.best_score_ > self.best_score:
+            self.best_score = clf.best_score_
+            self.best_estimator_params = clf.best_params_
+            self.best_mida_params = mida_params
+            self.best_transformer_params = transformer_params
+
+        return self
+
+    def predict(self, x, groups=None):
+        """Predict the labels for the given data x
+
+        Args:
+            x (array-like tensor): input data, shape (n_samples, n_features)
+            groups (array-like, optional): group labels for the samples, shape (n_samples,)
+
+        Returns:
+            array-like: Predicted labels, shape (n_samples, )
+        """
+        return self.best_estimator.predict(self.transform(x, groups=groups))
+
+    def decision_function(self, x, groups=None):
+        """Decision scores of each class for the given data x
+
+        Args:
+            x (array-like tensor): input data, shape (n_samples, n_features)
+            groups (array-like, optional): group labels for the samples, shape (n_samples,)
+
+        Returns:
+            array-like: decision scores, shape (n_samples,) for binary case, else (n_samples, n_class)
```

Comment:
Please ensure consistency in capitalising the word "decision" throughout this file.